src/classes/ExpectObject_Tests.cls

```diff
@@ -0,0 +1,78 @@
+@isTest
+public class ExpectObject_Tests {
+    @isTest
+    static void ExpectEqualObjectsShouldNotThrow() {
+        Object one = new List<Account>{new Account(Name = 'Hello', Phone = '1')};
+        Object two = new List<Account>{new Account(Phone = '1', Name = 'Hello')};
+
+        Expect.that(one).shouldEqual(two);
```

Comment:
For inspiration...

http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#custom-comparison-strategy
https://shouldjs.github.io/#should-equal
https://github.com/nunit/docs/wiki/CollectionEquivalentConstraint